src/js/providers/default.ts

```diff
@@ -1,9 +1,68 @@
 import { PLAYER_STATE, MEDIA_TYPE } from 'events/events';
+import { GenericObject, SourceObj, SeekRange } from 'types/generic.type';
+
+const noop: () => void = function(): void { /* noop */ };
+const returnFalse: () => boolean = (() => false);
+const getNameResult: { name: string } = { name: 'default' };
+const returnName: () => { name: string } = (() => getNameResult);
+interface DefaultProvider {
```

Comment:
What do you think about exporting this interface and making all providers implement it?

`export interface Provider`

Reply:
👍 yeah i like this